File: Java/test/LinkedListTester.java
import Lists.LinkedList;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LinkedListTester extends ListTester {
    public LinkedListTester() {
        super(new LinkedList<>(Integer.class));
    }

    @Test
    public void testGetFirst() {
        assertEquals(null, getLinkedList().getFirst());

        PopulateTestList();
        assertEquals(getLinkedList().get(0), getLinkedList().getFirst());
    }

    @Test
    public void testGetLast() {
        assertEquals(null, getLinkedList().getLast());

        PopulateTestList();
        assertEquals(getLinkedList().get(getLinkedList().size() - 1), getLinkedList().getLast());
    }

    @Test
    public void testAddFirst() {
        PopulateTestList();
        getLinkedList().addFirst(-1);
        assertEquals(-1, getLinkedList().get(0).intValue());

        getLinkedList().addFirst(500);
        assertEquals(500, getLinkedList().getFirst().intValue());
    }

    @Test
    public void testAddLast() {
        PopulateTestList();
        getLinkedList().addLast(-1);
        assertEquals(-1, getLinkedList().get(getLinkedList().size() - 1).intValue());

        getLinkedList().addLast(500);
        assertEquals(500, getLinkedList().getLast().intValue());
    }

    private LinkedList<Integer> getLinkedList() {
        return (LinkedList<Integer>) testList;
    }
}

File: CSharp/DataStructuresTester/LinkedListTester.cs
using DataStructures.Lists;
using DataStructuresTester.Fixtures;
using Xunit;

namespace DataStructuresTester
{
    public class LinkedListTester : ListTester, IClassFixture<LinkedListFixture<int?>>
    {
        private LinkedList<int?> TestLinkedList => (LinkedList<int?>)Fixture.TestList;

        public LinkedListTester(LinkedListFixture<int?> listFixture)
            : base(listFixture)
        {

        }

        [Fact]
        public void TestAddFirst()
        {
            PopulateTestList();
            TestLinkedList.AddFirst(-1);
            Assert.Equal(-1, TestLinkedList[0]);

            TestLinkedList.AddFirst(500);
            Assert.Equal(500, TestLinkedList.First);
        }

        [Fact]
        public void TestAddLast()
        {
            PopulateTestList();
            TestLinkedList.AddLast(-1);
            Assert.Equal(-1, TestLinkedList[^1]);

            TestLinkedList.AddLast(500);
            Assert.Equal(500, TestLinkedList.Last);
        }

        [Fact]
        public void TestRemoveFirst()
        {
            PopulateTestList();
            int? firstValue = TestLinkedList[0];
            TestLinkedList.RemoveFirst();
            Assert.DoesNotContain(firstValue, TestLinkedList);

            firstValue = TestLinkedList.First;
            TestLinkedList.RemoveFirst();
            Assert.DoesNotContain(firstValue, TestLinkedList);
        }

        [Fact]
        public void TestRemoveLast()
        {
            PopulateTestList();
            int? lastValue = TestLinkedList[^1];
            TestLinkedList.RemoveLast();
            Assert.DoesNotContain(lastValue, TestLinkedList);

            lastValue = TestLinkedList.Last;
            TestLinkedList.RemoveLast();
            Assert.DoesNotContain(lastValue, TestLinkedList);
        }
    }
}
